src/main/java/com/google/codeu/servlets/UFODataServlet.java

```diff
@@ -0,0 +1,54 @@
+package com.google.codeu.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Returns UFO data as a JSON array, e.g. [{"lat": 38.4404675, "lng": -122.7144313}]
+ */
+@WebServlet("/ufo-data")
+public class UfoDataServlet extends HttpServlet {
+
+    private JsonArray ufoSightingArray;
+
+    @Override
+    public void init() {
+        ufoSightingArray = new JsonArray();
+        Gson gson = new Gson();
+        Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/ufo-data.csv"));
+        while(scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] cells = line.split(",");
+
+            double lat = Double.parseDouble(cells[0]);
+            double lng = Double.parseDouble(cells[1]);
+
+            ufoSightingArray.add(gson.toJsonTree(new UfoSighting(lat, lng)));
+        }
+        scanner.close();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+        response.getOutputStream().println(ufoSightingArray.toString());
```

Comment:
What does this println do? Where does it print to?

Reply:
It prints the json.  if you go to localhost:8080/UfoDataServlet.java you will see the json representation of all the lat and long points.